src/main/java/pl/edu/agh/firecell/engine/algorithm/BasicAlgorithm.java

```diff
@@ -1,52 +1,126 @@
+
+        switch (oldCell.type()) {
+            case SOLID -> {
+                newTemperature = oldCell.temperature() + computeConductivityFromAll(oldState, oldCell, cellIndex);
 
-        double zTemp = computeConductivity(
-                oldState.getCell(IndexUtils.up(cellIndex)),
-                oldCell,
-                oldState.getCell(IndexUtils.down(cellIndex)));
+                if (!oldCell.flammable()) {
+                    break;
+                }
+                // is flammable
 
-        double xTemp = computeConductivity(
-                oldState.getCell(IndexUtils.east(cellIndex)),
-                oldCell,
-                oldState.getCell(IndexUtils.west(cellIndex)));
 
-        double averageTemp = DoubleStream.of(yTemp, zTemp, xTemp)
-                .average()
-                .orElseThrow(() -> new IllegalStateException("Empty stream while computing average temperature"));
+            }
+            case FLUID -> {
+                newTemperature = oldCell.temperature() + computeConvectionForFluid(oldState, oldCell, cellIndex);
+            }
+        }
 
-        return new Cell(averageTemp, oldCell.conductivityCoefficient());
+        return new Cell(
+                newTemperature,
+                newBurningTime,
+                newFlammable,
+                newType,
+                newMaterial
+        );
     }
 
     private double computeConductivity(Cell former, Cell middle, Cell latter) {
-
         // TODO: put them into class constants and give meaningful names
-        double dt = 0.05;
-        double gammaPrimN = 0;
-        double gammaPrimN1 = 0;
-
-        return middle.temperature() +
-                dt * (
-                        gammaPrimN * (middle.temperature() - former.temperature()) -
-                                gammaPrimN1 * (latter.temperature() - middle.temperature())
-                );
+
+        return deltaTime * (
+                GAMMA_PRIM_N1 * (middle.temperature() - former.temperature()) -
+                        GAMMA_PRIM_N2 * (latter.temperature() - middle.temperature())
+        );
+    }
+
+    private double computeConductivityFromAll(State oldState, Cell oldCell, Vector3i cellIndex) {
+
+        double yTemp;
+        double zTemp;
+        double xTemp;
+
+        try {
+
+            yTemp = computeConductivity(
+                    oldState.getCell(IndexUtils.north(cellIndex)),
+                    oldCell,
+                    oldState.getCell(IndexUtils.south(cellIndex)));
+
+            zTemp = computeConductivity(
+                    oldState.getCell(IndexUtils.up(cellIndex)),
+                    oldCell,
+                    oldState.getCell(IndexUtils.down(cellIndex)));
+
+            xTemp = computeConductivity(
+                    oldState.getCell(IndexUtils.east(cellIndex)),
+                    oldCell,
+                    oldState.getCell(IndexUtils.west(cellIndex)));
+
+        } catch (IllegalArgumentException e) {
```

Comment:
It's unnecessary to print this exception, since it actaully does not mean anything wrong.
If we cannot find one of the neighbours, conducivity in this direction should be computed another way or not icnluded at all.

Reply:
Is it ok to print it in debug level?